### ISSUE TITLE
[Feature]: Deprecate QueryErrorBoundary?

### ISSUE THREAD

**Issue by manudeli** [toss/suspensive#1325]
### Package Scope

@suspensive/react-query

### Description

```text
// optional-peer-dep:__vite-optional-peer-dep:@suspensive/react:@suspensive/react-query-4
throw new Error(`Could not resolve "@suspensive/react" imported by "@suspensive/react-query-4". Is it installed?`);
```

Due to the peerDep issue of @suspensive/react-query, some projects have the problem of having to install @suspensive/react. The implementation of QueryErrorBoundary is simple and leaving this implementation up to the user seems to end these problems and solve the peerDep problem.

@gwansikk I want to talk about this issue

**Comment by gwansikk** [comment]
> The implementation of QueryErrorBoundary is simple and leaving this implementation up to the user seems to end these problems and solve the peerDep problem.

I agree. A QueryErrorBoundary can be easily created by the user, and I believe it should be up to the user’s discretion as needed.

**Comment by manudeli** [comment]
Let's do this in v3
